provider/google.go

```diff
@@ -263,14 +244,71 @@ func (p *googleProvider) submitChange(zone string, change *dns.Change) error {
 		return nil
 	}
 
-	_, err := p.changesClient.Create(p.project, zone, change).Do()
+	zones, err := p.Zones()
 	if err != nil {
 		return err
 	}
 
+	// separate into per-zone change sets to be passed to the API.
+	changes := separateChange(zones, change)
+
+	for z, c := range changes {
+		if _, err := p.changesClient.Create(p.project, z, c).Do(); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+// separateChange separates a multi-zone change into a single change per zone.
+func separateChange(zones map[string]*dns.ManagedZone, change *dns.Change) map[string]*dns.Change {
```

Comment:
That would be great but would require to create a common object for zones and record(-lists), which I would like to defer to after v0.3.